backend/inspirehep/search/api.py

```diff
@@ -186,10 +186,9 @@ def execute(self, *args, **kwargs):
         return results
 
     def query_for_superuser_or_users(self, query_string):
+        collection = request.args.get("collection", default="literature", type=str)
         if not is_superuser_or_cataloger_logged_in():
-            user_query = Q(
-                IQ(query_string, self) & Q("term", _collections="Literature")
-            )
+            user_query = Q(IQ(query_string, self) & Q("term", _collections=collection))
```

Comment:
what prevents a non-priviledged user from searching in private collections now, by setting something like `collection=d0 internal notes`?